tests/events/tests_events_register_api.py

```diff
@@ -0,0 +1,116 @@
+from rest_framework.test import APITestCase, APIClient
+from rest_framework import status
+from events.models import Event
+from users.models import CustomUser
+
+class EventRegisterTestCase(APITestCase):
+    def setUp(self):
+        """
+        Create different events with different years and flag input in test database
+        """
+        # create an unverifeid authenticated user
+        self.auth_unverified_user_email = "crash.test1.dummy@gmail.com"
+        self.auth_unverified_user_password = "test1.modelx"
+        self.auth_unverified_user = CustomUser.objects.create_user(
+            email = self.auth_unverified_user_email,
+            username = self.auth_unverified_user_email,
+            password = self.auth_unverified_user_password,
+            verified=False
+        )
+        
+        # create an verified authenticated user
+        self.auth_verified_user_email = "crash.test2.dummy@gmail.com"
+        self.auth_verified_user_password = "test2.modelx"
+        self.auth_verified_user = CustomUser.objects.create_user(
+            email = self.auth_verified_user_email,
+            username = self.auth_verified_user_email,
+            password = self.auth_verified_user_password,
+            verified=True
+        )
+
+        login_api="/users/login/"
+        client = APIClient()
+        # get auth token of unverified user
+        unverified_login_payload = {
+            'email':self.auth_unverified_user_email,
+            'password':self.auth_unverified_user_password
+        }
+        unverified_login_response = client.post(login_api,unverified_login_payload)
+        self.unverified_auth_token = unverified_login_response.data['token']
```

Comment:
As `AuthAPITestCase` has been added in #42, please use it for creating an auth client ✌️